### PR TITLE
Checklist firebase database addition

### PR DESCRIPTION
**Pull request by zhiqiao**
Enables ChecklistActivity to pull data from Firebase Database.

### CODE REVIEW

File: app/src/main/java/com/g11x/checklistapp/data/ChecklistItem.java [hunk 1]
```diff
@@ -19,14 +19,22 @@
...
+@IgnoreExtraProperties
 public class ChecklistItem {
-  private final String name;
-  private final String location;
-  private final String description;
-  private final Uri directions;
+  public String name;
+  public String location;
+  public String description;
+  private Uri directions;
```
Comment (hiwaylon) [inline app/src/main/java/com/g11x/checklistapp/data/ChecklistItem.java:29]: Is there a particular reason this is different than the rest?

File: app/src/main/java/com/g11x/checklistapp/ChecklistActivity.java [hunk 2]
```diff
@@ -25,38 +25,100 @@
...
+
+      @Override
+      public void onChildMoved(DataSnapshot dataSnapshot, String s) {
+        // Do nothing, for now.  Should re-order list of items.
+      }
+
+      @Override
+      public void onCancelled(DatabaseError databaseError) {
+        Log.e("Firebase", "Could not read data from remote database.");
+        Toast.makeText(getApplicationContext(), "Failed database read", Toast.LENGTH_SHORT);
```
Comment (aaronj1335) [inline app/src/main/java/com/g11x/checklistapp/ChecklistActivity.java]: I think [Snackbars are slightly recommended](https://material.google.com/components/snackbars-toasts.html), and we're [using them elsewhere](https://github.com/g11x/checklistapp/blob/e56d35777e5368278b398541f641927c478513b9/app/src/main/java/com/g11x/checklistapp/ImportantInformationActivity.java#L76) currently.

**Comment by hiwaylon** [comment]
Had a minor non-blocking question, but otherwise 👍

File: app/src/main/java/com/g11x/checklistapp/ChecklistActivity.java [hunk 3]
```diff
@@ -25,38 +25,100 @@
...
-  private ChecklistAdapter adapter;
+  private ChecklistAdapter checklistAdapter;
+  private DatabaseReference checklistRef;
+  private final List<ChecklistItem> checklistItems = new ArrayList<>();
+  private ChildEventListener childEventListener;
+
+  private void addChildListeners(DatabaseReference reference) {
+    childEventListener = new ChildEventListener() {
+      @Override
+      public void onChildAdded(DataSnapshot dataSnapshot, String s) {
```
Comment (aaronj1335) [inline app/src/main/java/com/g11x/checklistapp/ChecklistActivity.java:54]: Any thoughts on using [FirebaseUI's adapter](https://github.com/firebase/FirebaseUI-Android/blob/master/database/README.md#using-firebaseui-to-populate-a-recyclerview)?

File: app/src/main/java/com/g11x/checklistapp/ChecklistActivity.java (hunk 2 above)
Comment (zhiqiao) [inline app/src/main/java/com/g11x/checklistapp/ChecklistActivity.java]: Done.

File: app/src/main/java/com/g11x/checklistapp/data/ChecklistItem.java (hunk 1 above)
Comment (zhiqiao) [inline app/src/main/java/com/g11x/checklistapp/data/ChecklistItem.java:29]: I just checked, members can be private for use by Firebase Database.

File: app/src/main/java/com/g11x/checklistapp/ChecklistActivity.java (hunk 3 above)
Comment (zhiqiao) [inline app/src/main/java/com/g11x/checklistapp/ChecklistActivity.java:54]: Will send follow-up CL since changes are extensive.

**Comment by zhiqiao** [comment]
Thanks Waylon and Aaron! PTAL.

**Comment by aaronj1335** [comment]
🚢 it

**Comment by zhiqiao** [comment]
Could you take another look? I've migrated to FirebaseRecyclerAdapter

**Comment by aaronj1335** [comment]
🚢 it

**Pull request merged**